From 694486dac679f4ddb93debfcd72061542d880bb9 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Tue, 15 Oct 2019 04:44:04 -0400
Subject: [PATCH] mark functions with address taken via assembly

This fixes compatibility with CFI in a build with !CONFIG_MODULES.

Signed-off-by: anupritaisno1 <www.anuprita804@gmail.com>
---
 arch/arm64/kernel/cpu_errata.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/kernel/cpu_errata.c b/arch/arm64/kernel/cpu_errata.c
index b9faf03b5736..274a4488a000 100644
--- a/arch/arm64/kernel/cpu_errata.c
+++ b/arch/arm64/kernel/cpu_errata.c
@@ -19,6 +19,7 @@
 #include <linux/arm-smccc.h>
 #include <linux/psci.h>
 #include <linux/types.h>
+#include <asm/alternative.h>
 #include <asm/cachetype.h>
 #include <asm/cpu.h>
 #include <asm/cputype.h>
@@ -275,6 +276,12 @@ void __init arm64_update_smccc_conduit(struct alt_instr *alt,
 	*updptr = cpu_to_le32(insn);
 }
 
+__attribute__((used))
+alternative_cb_t arm64_update_smccc_conduit_indirectly_callable(void)
+{
+	return &arm64_update_smccc_conduit;
+}
+
 void __init arm64_enable_wa2_handling(struct alt_instr *alt,
 				      __le32 *origptr, __le32 *updptr,
 				      int nr_inst)
@@ -289,6 +296,12 @@ void __init arm64_enable_wa2_handling(struct alt_instr *alt,
 		*updptr = cpu_to_le32(aarch64_insn_gen_nop());
 }
 
+__attribute__((used))
+alternative_cb_t arm64_enable_wa2_handling_indirectly_callable(void)
+{
+	return &arm64_enable_wa2_handling;
+}
+
 void arm64_set_ssbd_mitigation(bool state)
 {
 	switch (psci_ops.conduit) {
